### PR TITLE
Move testing stuff to common

### PR DESCRIPTION
**Pull request by talIguaz**

### CODE REVIEW

File: pkg/tsdb/tsdbtest/config.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+)
+
+const TsdbDefaultTestConfigPath = "testdata"
+const relativeProjectPath = "src/github.com/v3io/v3io-tsdb"
+
+/*
+This method will try and load the configuration file from several locations by the following order:
+1. Current package's 'testdata/v3io.yaml' folder
+2. $GOPATH/src/github.com/v3io/v3io-tsdb/v3io.yaml
+3. Environment variable named 'CONF'
```
Comment (gtopper) [inline pkg/tsdb/tsdbtest/config.go]: This is probably too generic a name.

File: pkg/tsdb/tsdbtest/config.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import (
+	"github.com/pkg/errors"
+	"github.com/v3io/v3io-tsdb/pkg/config"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const TsdbDefaultTestConfigPath = "testdata"
+const relativeProjectPath = "src/github.com/v3io/v3io-tsdb"
```
Comment (gtopper) [inline pkg/tsdb/tsdbtest/config.go]: Are these two each purposefully public and private?

File: pkg/tsdb/testdata/v3io.yaml.template [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+workers: 20
+
+username: <user name>
+password: <password>
+
+performance:
+  reportOnShutdown: true
+  output: "stdout"
+  reportPeriodically: true
+  reportInterval: 30
```
Comment (gtopper) [inline pkg/tsdb/testdata/v3io.yaml.template]: Newline missing.

File: pkg/tsdb/tsdbtest/config.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+/*
+This method will try and load the configuration file from several locations by the following order:
+1. Current package's 'testdata/v3io.yaml' folder
+2. $GOPATH/src/github.com/v3io/v3io-tsdb/v3io.yaml
```
Comment (gtopper) [inline pkg/tsdb/tsdbtest/config.go]: Should handle complex GOPATH as well.

File: pkg/tsdb/testdata/v3io.yaml.template [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+# Path in the container (default, should be override by test)
+path: <tsdb path>
+
+# Logging level. Valid values: debug,info,warn,error (Default: info)
+verbose: "warn"
+
+# Default timeout 10 seconds
+timeout: 30
+
+workers: 20
```
Comment (imakhlin) [inline pkg/tsdb/testdata/v3io.yaml.template]: 8

File: pkg/tsdb/testdata/v3io.yaml.template [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+username: <user name>
+password: <password>
+
+performance:
```
Comment (imakhlin) [inline pkg/tsdb/testdata/v3io.yaml.template]: remove it

File: pkg/tsdb/tsdbtest/config.go [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+3. Environment variable named 'V3IO_CONF'
+*/
+func GetV3ioConfigPath() (string, error) {
```
Comment (imakhlin) [inline pkg/tsdb/tsdbtest/config.go]: Tests are missing

File: pkg/tsdb/tsdbtest/tsdbtest.go [hunk 8]
```diff
@@ -31,24 +40,29 @@ func DeleteTSDB(t *testing.T, v3ioConfig *config.V3ioConfig) {
 	}
 }
 
-func SetUp(t *testing.T, v3ioConfig *config.V3ioConfig) func() {
+func CreateTestTSDB(t testing.TB, v3ioConfig *config.V3ioConfig) {
 	dbConfig := config.DBPartConfig{
 		DaysPerObj:     1,
 		HrInChunk:      1,
-		DefaultRollups: "sum",
+		DefaultRollups: "*",
```
Comment (imakhlin) [inline pkg/tsdb/tsdbtest/tsdbtest.go]: Make sure it works

File: pkg/tsdb/tsdbtest/tsdbtest.go [hunk 9]
```diff
@@ -31,24 +40,29 @@ func DeleteTSDB(t *testing.T, v3ioConfig *config.V3ioConfig) {
...
 		RollupMin:      10,
 	}
-	v3ioConfig.Path = fmt.Sprintf("%s-%d", v3ioConfig.Path, time.Now().Nanosecond())
 	if err := CreateTSDB(v3ioConfig, &dbConfig); err != nil {
 		t.Fatalf("Failed to create TSDB. reason: %s", err)
 	}
+}
+
+func SetUp(t testing.TB, v3ioConfig *config.V3ioConfig) func() {
+	v3ioConfig.Path = fmt.Sprintf("%s-%d", v3ioConfig.Path, time.Now().Nanosecond())
```
Comment (imakhlin) [inline pkg/tsdb/tsdbtest/tsdbtest.go]: see test/benchmark/common/utils.go:62

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 10]
```diff
@@ -84,6 +85,10 @@ func testIngestDataCase(t *testing.T, v3ioConfig *config.V3ioConfig,
 	metricsName string, userLabels []utils.Label, data []tsdbtest.DataPoint) {
 	defer tsdbtest.SetUp(t, v3ioConfig)()
 
+	sort.Sort(tsdbtest.DataPointTimeSorter(data))
```
Comment (gtopper) [inline pkg/tsdb/v3iotsdb_integration_test.go]: I don't think the type conversion does anything here.

**Pull request merged**